libs/api/domains/driving-license/src/lib/graphql/qualityPhoto.resolver.ts

```diff
@@ -0,0 +1,32 @@
+import { Query, Resolver, ResolveField, Parent } from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+import { ApiScope } from '@island.is/auth/scopes'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  IdsUserGuard,
+  ScopesGuard,
+  CurrentUser,
+  Scopes,
+} from '@island.is/auth-nest-tools'
+import { DrivingLicenseService } from '../drivingLicense.service'
+export * from '@island.is/nest/audit'
+import { QualityPhoto } from './models'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Resolver(QualityPhoto)
+export class QualityPhotoResolver {
+  constructor(private readonly drivingLicenseService: DrivingLicenseService) {}
+
+  @ResolveField('qualityPhotoDataUri', () => String, { nullable: true })
```

Comment:
The usage of the field resolver can be improved.
So both the parent resolver (`drivingLicenseQualityPhoto`) and the field resolver are performing a service call that **BOTH** make the same API call:
```typescript
const image = await this.drivingLicenseApi.getHasQualityPhoto({
  nationalId,
})
 ```

So under the hood, a single GraphQL query is making two identical API calls.

It would be better if the field resolver would just check if the parent is `null` and skip that check.